src/main/java/net/tylers1066/beaming/sign/CrewSign.java

```diff
@@ -76,6 +77,28 @@ public final void onSignRightClick(@NotNull PlayerInteractEvent event) {
         u.setHome("home", location);
     }
 
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+    public void onSignTranslate(@NotNull SignTranslateEvent event) {
+        if (!Config.UpdateBedLocations || !Utils.isCrewSign(event::getLine))
+            return;
+
+        World w = event.getCraft().getWorld();
+        for (MovecraftLocation l : event.getLocations()) {
+            Location signLocation = l.toBukkit(w);
+
+            Block bed = Utils.getBedBlock(signLocation);
+            if (!Utils.isBed(bed))
+                continue;
+
+            Player player = Bukkit.getPlayerExact(event.getLine(1));
+            if (player == null) // cannot change the bed locations of offline players
+                continue;
+
+            player.setBedSpawnLocation(bed.getLocation());
+            break; // stop after the first valid sign placement
+        }
+    }
+
```

Comment:
Wouldn't it make more sense to only run this on craft release?  The rest of the system should make the player respawn on any beds of their piloted craft.

Reply:
I say this because `SignTranslateEvent` has a *very heavy* performance cost, and we've been hoping to look for ways to remove it's usage everywhere.

Reply:
I was considering that, I'll see if I can make it work.